pkg/operator/workload/workload_openshiftapiserver_v311_00_sync_test.go

```diff
@@ -94,20 +99,33 @@ func TestOperatorConfigProgressingCondition(t *testing.T) {
 				},
 			}
 			apiServiceOperatorClient := operatorfake.NewSimpleClientset(operatorConfig)
-			openshiftConfigClient := configfake.NewSimpleClientset(&configv1.Image{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}})
+			openshiftConfigClient := configfake.NewSimpleClientset(
+				&configv1.Image{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}},
+				&configv1.ClusterVersion{ObjectMeta: metav1.ObjectMeta{Name: "version"}},
+			)
+
 			fakeOperatorClient := operatorv1helpers.NewFakeOperatorClient(&operatorv1.OperatorSpec{ManagementState: operatorv1.Managed}, &operatorv1.OperatorStatus{}, nil)
+			configInformers := configinformers.NewSharedInformerFactory(openshiftConfigClient, 10*time.Minute)
 
 			target := OpenShiftAPIServerWorkload{
 				kubeClient:                kubeClient,
 				operatorClient:            fakeOperatorClient,
 				operatorConfigClient:      apiServiceOperatorClient.OperatorV1(),
 				openshiftConfigClient:     openshiftConfigClient.ConfigV1(),
+				clusterVersionInformer:    configInformers.Config().V1().ClusterVersions(),
 				versionRecorder:           status.NewVersionGetter(),
 				countNodes:                fakeCountNodes,
 				ensureAtMostOnePodPerNode: func(spec *appsv1.DeploymentSpec, componentName string) error { return nil },
 			}
 
-			if _, _, err := target.Sync(context.Background(), factory.NewSyncContext("TestSyncCOntext", events.NewInMemoryRecorder(""))); len(err) > 0 {
+			// register the informer
+			configInformers.Config().V1().ClusterVersions().Informer()
+
+			ctx := context.Background()
+			configInformers.Start(ctx.Done())
```

Comment:
and having a lister instead of an informer will simplify this unit tests :)

Reply:
Done